datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -304,10 +304,11 @@ async fn not_between_integers() -> Result<()> {
     .await
 }
 
-#[tokio::test]
-async fn simple_scalar_function_abs() -> Result<()> {
-    roundtrip("SELECT ABS(a) FROM data").await
-}
+// TODO: disable this test until the support for UserDefinedScalarFunction
+// #[tokio::test]
+// async fn simple_scalar_function_abs() -> Result<()> {
+//     roundtrip("SELECT ABS(a) FROM data").await
+// }
```

Comment:
I disabled this test case because now it seems impossible for substrait to get a ScalarFunction from name. See: https://github.com/apache/arrow-datafusion/blob/91f3eb2e5430d23e2b551e66732bec1a3a575971/datafusion/substrait/src/logical_plan/consumer.rs#L116-L130 and https://github.com/apache/arrow-datafusion/blob/91f3eb2e5430d23e2b551e66732bec1a3a575971/datafusion/substrait/src/logical_plan/consumer.rs#L77-L82
I can try to implement this function in this PR or as a follow-up.

Reply:
Also @SteveLauC, I think you also may experience this question also. Cause no test case for `acos` so CI could work now.

Reply:
I think we may need to fix the issue beforehand to avoid regression on this test

Reply:
filed https://github.com/apache/arrow-datafusion/issues/9336

Reply:
Fixed

Reply:
Thanks! @jayzhan211 
I applied the changes and this PR should work now.